packages/server/src/api/controllers/user.ts

```diff
@@ -25,6 +27,12 @@ export async function syncUser(ctx: BBContext) {
       throw err
     }
   }
+
+  let previousApps =
+    previousUser && isUser(previousUser)
```

Comment:
We can remove the need for the `isUser` here by using parameters on the `ctx: Ctx`. 

Now that the sync user request has a body, we can add a new type here: https://github.com/Budibase/budibase/blob/develop/packages/types/src/api/web/user.ts

Like:

```ts
export interface SyncUserRequest {
   previousUser: User
}
```

we can then update the function signature to look like:
```ts
export async function syncUser(ctx: Ctx<SyncUserRequest>) {
```

Now our `ctx.request.body` will automatically infer the `previousUser` as a `User` object